src/flattenizer.ts

```diff
@@ -15,7 +15,7 @@
  * @throws {TypeError}             - if object passed in is not an object or if the delimiter is not a string
  * @public
  */
-export const flatten = (unflattened: object, delimiter: string | null | undefined = '.'): object => {
+export const flatten = (unflattened: object, delimiter: string = '.'): object => {
   if (unflattened === undefined || unflattened === null) { return unflattened; }
```

Comment:
✏️ I'm surprised the compiler let's this through. `unflattened` should always be an `object` according to the signature, and this function should always return an `object`, but this line allows for passing `nil`ish thing and returning it.

Suggestion: `unflattened: object = {}` or is that too opinionated?

Reply:
I also had trouble with this. For backwards compatibility I wanted to keep this. Also, people don't all use typescript, so I feel like it should still be there.

Reply:
I'm not sure, I'll take thoughts and suggestions. I'll also be releasing a major bump, so I do have freedom.